pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -354,6 +383,33 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 	return dataFromTemplate, nil
 }
 
+func convertRegistryMapToSliceInOrder(registryMap map[int]map[string]interface{}) ([]map[string]interface{}, error) {
+	size := len(registryMap)
+	registries := make([]map[string]interface{}, size)
+	for k, v := range registryMap {
+		if k >= size {
+			return nil, fmt.Errorf("the index %d is out of the bound of the registry list (size of %d)", k, size)
+		}
+		registries[k] = v
+		registries[k][managementv3.PrivateRegistryFieldIsDefault] = false
+	}
+	// set the first registry in the list as the default registry
+	registries[0][managementv3.PrivateRegistryFieldIsDefault] = true
+	return registries, nil
+}
+
+func getIndexFromQuestion(question string) (int, error) {
+	re, err := regexp.Compile(`\[\d+\]`)
```

Comment:
I would define this as a var above the function with regex must compile.  The fewer error cases to consiter the better.

Reply:
We have had multiple conversations on using whether `regexp.Compile` or `regexp.MustCompile` in this PR and decided to go with the current implementation.  

I like your suggestion about defining the expression as a var, but I will keep it as it is now to avoid another round of approval.